analyzers/src/SonarAnalyzer.Common/Rules/UnconditionalJumpStatementBase.cs

```diff
@@ -19,29 +19,36 @@
  */
 
 using System.Collections.Generic;
+using System.Collections.Immutable;
 using System.Linq;
 using Microsoft.CodeAnalysis;
 using Microsoft.CodeAnalysis.Diagnostics;
 using SonarAnalyzer.Helpers;
 
 namespace SonarAnalyzer.Rules
 {
-    public abstract class UnconditionalJumpStatementBase<TStatementSyntax, TLanguageKindEnum> : SonarDiagnosticAnalyzer
+    public abstract class UnconditionalJumpStatementBase<TStatementSyntax, TSyntaxKind> : SonarDiagnosticAnalyzer
         where TStatementSyntax : SyntaxNode
-        where TLanguageKindEnum : struct
+        where TSyntaxKind : struct
     {
         protected const string DiagnosticId = "S1751";
-        protected const string MessageFormat = "Refactor the containing loop to do more than one iteration.";
+        private const string MessageFormat = "Refactor the containing loop to do more than one iteration.";
 
-        protected abstract GeneratedCodeRecognizer GeneratedCodeRecognizer { get; }
+        private readonly DiagnosticDescriptor rule;
 
-        protected abstract LoopWalkerBase<TStatementSyntax, TLanguageKindEnum> GetWalker(SyntaxNodeAnalysisContext context);
+        protected abstract ILanguageFacade<TSyntaxKind> Language { get; }
```

Comment:
This and the next abstract property `LoopStatements` should be together. And they won't need empty line between them.